File: Leet 3876/solution.py
def uniform_array(nums1: list[int]) -> bool:
    min_odd: int = -1
    for n in nums1:
        if n % 2 == 1:
            if min_odd == -1 or n < min_odd:
                min_odd = n

    # even - odd = odd
    # that odd should be >= 1
    for n in nums1:
        if n % 2 == 0 and n - min_odd < 1:
            return False
            
    return True


def main() -> None:
    print("3876. Construct Uniform Parity Array II")

    result: str = "pass" if uniform_array([1,4,7]) else "fail"
    print(f"[1,4,7] should equal true ? {result}")


if __name__ == "__main__":
    main()

File: Leet 3876/solution.cs
class Solution
{
    public bool UniformArray(int[] nums1)
    {
        int minOdd = -1;
        foreach(int num in nums1)
        {
            if (num % 2 == 1)
            {
                if (minOdd == -1 || num < minOdd)
                {
                    minOdd = num;
                }
            }
        }

        // Rules
        // even - odd = odd
        // nums[i] - nums[j] should be >= 1
        foreach(int num in nums1)
        {
            if (num % 2 == 0 && num - minOdd < 1)
            {
                return false;
            }
        }

        return true;
    }
}

class Program
{
    private static void Main()
    {
        Console.WriteLine("3876. Construct Uniform Parity Array II");

        var solution = new Solution();

        int[] nums1 = [1,4,7];
        string result = solution.UniformArray(nums1) ? "pass":"fail";
        Console.WriteLine($"[1,4,7] should equal true? {result}");
    }
}